vme/src/dilexp.cpp

```diff
@@ -2059,50 +1983,22 @@ void dilfe_filesz(dilprg *p)
 {
     dilval *v = new dilval;
     dilval *v1 = p->stack.pop();
-    char filename[511];
-    FILE *fp = nullptr;
 
     v->type = DILV_INT;
 
     switch (dil_getval(v1))
     {
         case DILV_SP:
-            if (str_is_empty(p->frame[0].tmpl->zone->getDILFilePath()))
-            {
-                if (!file_exists(g_cServerConfig.getDILFileDir()))
-                {
-#ifdef _WINDOWS
-                    _mkdir(g_cServerConfig.m_dilfiledir);
-#else
-                    mkdir(g_cServerConfig.getDILFileDir().c_str(), S_IRUSR | S_IWUSR | S_IXUSR | S_IRGRP | S_IWGRP | S_IXGRP);
-#endif
-                }
-                strcpy(filename, g_cServerConfig.getDILFileDir().c_str());
-            }
-            else
-            {
-                if (!file_exists(p->frame[0].tmpl->zone->getDILFilePath()))
-                {
-#ifdef _WINDOWS
-                    _mkdir(p->frame[0].tmpl->zone->getDILFilePath());
-#else
-                    mkdir(p->frame[0].tmpl->zone->getDILFilePath(), S_IRUSR | S_IWUSR | S_IXUSR | S_IRGRP | S_IWGRP | S_IXGRP);
-#endif
-                }
-
-                strcpy(filename, p->frame[0].tmpl->zone->getDILFilePath());
-            }
-            strcat(filename, "/strings/");
-            if (!file_exists(filename))
+        {
+            std::filesystem::path filename{p->frame[0].tmpl->zone->getDILFilePath().value_or(g_cServerConfig.getDILFileDir())};
```

Comment:
These are all super nice but adding tests for them would be nice :-) But I'll close my eyes and try to just run it :-D

Reply:
What were you thinking to test? I assume std::optional and std::filesystem stuff work.
It could be even neater because the `operator/` actually adds system specific directory separators as well, its kinda cool https://en.cppreference.com/w/cpp/filesystem/path/operator_slash